### PR TITLE
Phpdoc cleanup

### PR DESCRIPTION
**Pull request by jwilson8767**
Most of the changes here represent phpdoc-only changes. However, there are few places where return types did not match what was in the phpdoc or seemed to be expected by the usages of the relevant method. I have done my best to ensure that all return types in the files I touched are now consistent with the way the return values are used.

Phpstorm was used to validate phpdoc blocks. 

At your request I can also clean up other files as part of this pr. I only cleaned the files immediately associated with a form issue I was troubleshooting today.

### CODE REVIEW

File: modules/backend/widgets/Form.php [hunk 1]
```diff
@@ -338,7 +353,7 @@ public function onRefresh()
                 if (!isset($this->allFields[$field])) {
                     continue;
                 }
-
+                /** @var FormWidgetBase $fieldObject */
```
Comment (jwilson8767) [inline modules/backend/widgets/Form.php]: Inline `@var` tags like this allow IDEs to resolve the method used on objects (and therefore also their return types)

File: modules/system/traits/ConfigMaker.php [hunk 2]
```diff
@@ -149,11 +153,11 @@ public function getConfigPath($fileName, $configPath = null)
         foreach ($configPath as $path) {
             $_fileName = $path . '/' . $fileName;
             if (File::isFile($_fileName)) {
-                break;
+                return $_fileName;
             }
         }
 
-        return $_fileName;
+        return '';
```
Comment (jwilson8767) [inline modules/system/traits/ConfigMaker.php]: This change prevents returning `$_fileName` which may be undefined.

File: modules/backend/widgets/Form.php [hunk 3]
```diff
@@ -526,17 +545,18 @@ public function addFields(array $fields, $addToArea = null)
...
     public function addTabFields(array $fields)
     {
-        return $this->addFields($fields, 'primary');
+        $this->addFields($fields, 'primary');
     }
 
     public function addSecondaryTabFields(array $fields)
     {
-        return $this->addFields($fields, 'secondary');
+        $this->addFields($fields, 'secondary');
```
Comment (jwilson8767) [inline modules/backend/widgets/Form.php]: `$this->addFields` returns void, and should not be returned.

**Comment by daftspunk** [comment]
Merged in 6f3d7a58f2d18e834a5d6cf51bc1702ac5bef117

Thanks for this, a quick note: we won't be using phpDoc for API docs, more than likely it will be a custom solution.

**Pull request closed without merging**